core/src/main/scala/org/apache/spark/ui/JettyUtils.scala

```diff
@@ -208,6 +210,13 @@ private[spark] object JettyUtils extends Logging {
         rewrittenURI.toString()
       }
 
+      override def newHttpClient(): HttpClient = {
+        // Use the Jetty logic to calculate the number of selector threads (#CPUs/2),
+        // but limit it to 8 max.
+        val numSelectors = math.max(1,math.min(8,Runtime.getRuntime().availableProcessors()/2))
```

Comment:
Aargh - what an unnecessary failure, sorry about that :(.
I've corrected the line in question, the Scala style tests do pass now.